### ISSUE TITLE
Can you recommend any tools for getting the most out of Box Drawing?

### ISSUE THREAD

**Issue by phyllisstein** [fabrizioschiavi/pragmatapro#169]
This is not by a mile a bug, but I’m curious whether folks familiar with PragmataPro’s box-drawing characters have any recommendations for helpful tools and a good place to start.

For context, I do a ton of diagramming. For work, it’s mostly software architectures, data flows, user journeys, and other engineering ephemera. Off the clock, I spend a lot of time on nonfiction writing (technical and otherwise). In this context, I rely pretty heavily on mapping---the only compositional “trick” I’ve found that isn’t deeply, profoundly silly.

To date, I’ve been able to get a ton of mileage out of [Monodraw](https://monodraw.helftone.com/) for software stuff. It’s more or less a WYSYWIG for generating diagram layouts from Unicode’s spec-standard box-drawing characters.

![CleanShot 2020-04-25 at 23 25 48 2020-04-26 08_10_19](https://user-images.githubusercontent.com/2207980/80307164-6b66d680-8795-11ea-8165-122d7e66ad4e.gif)

For mind mapping, I’ve leaned heavily on the lovely and talented [MindNode](https://mindnode.com/) for years.

![](https://garb.ag/safex.png)

Again, this is a snapshot of what I do now and not a feature request. Both tools are great because they do a lot for you; but the downside is, they do a lot for you.

The box-drawing samples in this repo have been hovering in the back of my mind as potential alternatives: they’re extremely lovely, and I’m drawn to their quality of being a little more austere than what I’m working with now, but still rich and expressive.

What seems unlovely is the prospect of managing layout with my spacebar.

![](https://garb.ag/pp-bd-oops.png)

So I’m curious whether anyone has managed to hack together a set of tools that exposes PragmataPro’s drawing glyphs and helps to manage some of the layout concerns around using them.

One possibility I’m looking at is adding a PragmataPro-friendly character palette and snippet set to Monodraw. It would be a little easier if I could find a canonical list of descriptive names for PragmataPro’s character set, since Monodraw’s custom palettes are JSON documents with description keys and glyph values. I’m digging through the resources in this repo in hopes of finding one as we speak.

I’ll certainly open a pull request if/when I can put that together. In the meantime, I’m deeply open to recommendations. 🥺

**Comment by fabrizioschiavi** [comment]
Good idea! If you can send me a Monodraw JSON document I can try to add description keys and glyphs values. Thanks for purpose me this step forward!

**Comment by phyllisstein** [comment]
As far as Monodraw goes, it turns out I misunderstood the spec. The JSON keys were _all_ `char`, and their values were characters---no human-readable name needed. I opened #170 with JSON documents defining custom Monodraw palettes for Pragmata Pro’s extended box drawing glyphs and its arrow sets.

I reproduced your Milan metro map to get a feel for whether and how it works. It’s not perfect, since you wind up having to draw character by character, but it beats the hell out of hard-spacing a layout into TextEdit. Unless you have other thoughts on favorite tools, this can probably be closed once the PR merges.
